core/src/main/java/org/pentaho/di/core/row/ValueDataUtil.java

```diff
@@ -853,25 +854,48 @@ public static Long divideLongs( Long a, Long b ) {
     return new Long( a.longValue() / b.longValue() );
   }
 
-  public static BigDecimal divideBigDecimals( BigDecimal a, BigDecimal b, MathContext mc ) {
-    BigDecimal result = null;
+  private static RoundingMode toRoundingMode( String roundingMode ) {
+    switch ( roundingMode ) {
+      case "CEILING":
+        return RoundingMode.CEILING;
+      case "DOWN":
+        return RoundingMode.DOWN;
+      case "FLOOR":
+        return RoundingMode.FLOOR;
+      case "HALF_DOWN":
+        return RoundingMode.HALF_DOWN;
+      case "HALF_EVEN":
+        return RoundingMode.HALF_EVEN;
+      case "HALF_UP":
+        return RoundingMode.HALF_UP;
+      case "UNNECESSARY":
+        return RoundingMode.UNNECESSARY;
+      case "UP":
+        return RoundingMode.UP;
+      default:
+        return RoundingMode.HALF_EVEN;
+    }
+  }
+
+  public static BigDecimal divideBigDecimals( BigDecimal a, BigDecimal b ) {
+    int precision = Integer.parseInt( System.getProperties()
```

Comment:
1- Instead of using System.getProperties() here, shouldn't use Kettle properties ?
2- We cannot delete the method that receives a MathContext as it's a public static method (users / partners / community may rely on it)
3- Suggest the default MathContext gets created in the static initializer and then used when MathContext comes in as null in divideBigDecimals.

Reply:
Now the default MathContext is being created in a static initializer.

Reply:
@mbatchelor I usually retrieve Kettle Properties this way... 
What are you thinking specifically? :)

Reply:
I updated the PR. Can you take a look @mbatchelor? :)